src/main.rs

```diff
@@ -69,7 +70,27 @@ where
                 anyhow::anyhow!("parsing ${name}: {err:#}")
             })?))
         }
-        Err(std::env::VarError::NotPresent) => Ok(None),
+        Err(std::env::VarError::NotPresent) => {
+            let secret_env_name = format!("{}_FILE", name);
+
+            match std::env::var(&secret_env_name) {
+                Ok(path) => {
+                    let content = std::fs::read_to_string(&path).with_context(|| {
+                        format!(
+                            "Reading secret for {name} from path defined in {secret_env_name}: {path}"
+                        )
+                    })?;
+
+                    let trimmed_content = content.trim();
```

Comment:
If the secret legitimately starts with a string (I know! What kind of a monster....!?), `.trim()` will break it.

`trim_end()` is probably a bit better in that regard.

```suggestion
                    let trimmed_content = content.trim_end();
```

that could still break a secret that legitimately ends with a space, so maybe it is even better to do something like:

```rust
content.trim_end_matches(|c| c == '\r' || c == '\n')
```

so that the trim only removes newlines?

Reply:
Good suggestion! `.triim_end()` is totally sufficient I believe - also this is just a little formatting help, in the end please use correct secrets in your secrets files. I have multiple other containers where I *need* to add a `\n` in the end for the containers to read the secret or exactly the other way around and have no `\n` in the end.